tests/tracer/test_encoders.py

```diff
@@ -435,7 +435,7 @@ def test_span_link_v04_encoding():
             SpanLink(trace_id=3, span_id=4, flags=0),
             SpanLink(
                 trace_id=(123 << 64) + 456,
-                span_id=2,
+                span_id=6,
```

Comment:
With this PR we no longer allow duplicate span links to be sent to the agent (span link is defined by a unique span_id <---> span_id pair). 

We need to change the span id of this link so it creates a unique link

Reply:
OK gotcha, so this previously was fine because we captured any link as a unique link (same span ID, but diff trace ID), but now we're overwriting if the span ID is a duplicate.